tests/unittests/sources/test_azure.py

```diff
@@ -1313,84 +1305,83 @@ def test_not_ds_detect_seed_should_return_no_datasource(self):
             assert self.m_read_dmi_data.mock_calls == [
                 mock.call("chassis-asset-tag")
             ]
-            self.assertFalse(ret)
+            assert not ret
             # Assert that for non viable platforms,
             # there is no communication with the Azure datasource.
-            self.assertEqual(0, m_crawl_metadata.call_count)
-            self.assertEqual(0, m_report_failure.call_count)
+            assert 0 == m_crawl_metadata.call_count
+            assert 0 == m_report_failure.call_count
 
-    def test_platform_viable_but_no_devs_should_return_no_datasource(self):
+    def test_platform_viable_but_no_devs_should_return_no_datasource(
+        self, get_ds, mocker
+    ):
         """For platforms where the Azure platform is viable
         (which is indicated by the matching asset tag),
         the absence of any devs at all (devs == candidate sources
         for crawling Azure datasource) is NOT expected.
-        Report failure to Azure as this is an unexpected fatal error.
+        Report failure to Azure as this is an unexpected error.
         """
         data = {}
-        dsrc = self._get_ds(data)
+        dsrc = get_ds(data)
+        mocker.patch(MOCKPATH + "list_possible_azure_ds", return_value=[])
         with mock.patch.object(dsrc, "_report_failure") as m_report_failure:
-            ret = dsrc.get_data()
-            self.assertFalse(ret)
-            self.assertEqual(1, m_report_failure.call_count)
+            assert dsrc.get_data() is True
```

Comment:
The reason why get_data was failing fatally was due to the subp error that is raised while unit tests by cloud-init test harness. Nowadays, the azure ds falls back to IMDS metadata.